### PR TITLE
C8 Updates

### PR DESCRIPTION
**Pull request by droberts-sea**
Changes include:
- Seeds:
    - Add a `seeds.rb` script to populate the DB (should cut a big piece of difficulty)
    - Clean up the seed data a bit based on issues we saw last time
    - Add a bit to the README about seeds
- User stories:
    - Attempt to streamline the wording and structure of the stories
    - Add hard requirements around reporting validation failures
    - Add more context about how much validation they should be doing
    - Simplify the trip creation / rating workflow (removed requirement that a passenger can have only one outstanding trip)
- README in general:
    - Add a stage level
    - Add a description of how work should be prioritized WRT learning goals
    - Lots of general cleanup

When looking at the README, the line diff makes it look like I rewrote the whole file. I found GitHub's rich diff to be more informative.

### CODE REVIEW

**Comment by droberts-sea** [comment]
I have run through the process of:
- Clone the project from scratch
- Run `rails new .`
- Generate some models
- Make sure the seed script works

So I have a high degree of confidence the seed file will work for students.

File: README.md [hunk 1]
```diff
@@ -31,71 +35,112 @@ Before you start writing _any_ code:
...
+- These models are related in a certain way
+- The field names of these models match the column names in the CSV files
+
+This is the recommended way to configure this project, but not a hard requirement. If you've got a good reason to change the setup, feel free to edit `db/seeds.rb` accordingly.
+
+### Things to Keep in Mind
+
+1. Use git relentlessly
+1. Pair program on difficult problems and to share learning experiences
+1. Deploy early (Wednesday), deploy often.
```
Comment (kariabancroft) [inline README.md]: Maybe better to just say at least "two days before submission"? So we don't have to update it if the week is different?

File: README.md [hunk 2]
```diff
@@ -31,71 +35,112 @@ Before you start writing _any_ code:
...
 
 ## User Stories
 
+This project has a lot of user stories, and it is likely you and your partner won't get to all of them. Here is our prioritized list of what's important for this project, based on this week's learning goals:
+
+1. CRUD operations on models with relations
+    - Creating new trips
+1. Model validations and reporting errors to the user
+1. Deployment
+1. Basic layout of a complex Rails app
```
Comment (kariabancroft) [inline README.md]: Layout might get confused with HTML/CSS - could we adjust to "structure" instead?

File: db/seeds.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+DRIVER_FILE = Rails.root.join('db', 'seed_data', 'drivers.csv')
+puts "Loading raw driver data from #{DRIVER_FILE}"
+
+CSV.foreach(DRIVER_FILE, :headers => true) do |row|
+  driver = Driver.new
+  driver.id = row['id']
+  driver.name = row['name']
+  driver.vin = row['vin']
+  puts "Created driver: #{driver.inspect}"
+  driver.save!
```
Comment (kariabancroft) [inline db/seeds.rb]: I really like when we have a seed file that keeps track of how many records have been added successfully and how many failed. I think this is a good practice to show the students. Any opposition to updating this to be a bit more forgiving if errors occur upon save?

**Review by kariabancroft — COMMENTED** [review]
Thanks for making these updates Dan - they look great and clarify the requirements. Do you feel like with the prioritization of learning goals there is enough clarity on which things they should avoid without having some explicit optional requirements?

**Review by tildeee — COMMENTED** [review]
Yup, the Readme updates read well to me, and the seeds script works on my end. To push forward on this, @kariabancroft , do you mind if I push to this PR?

**Comment by kariabancroft** [comment]
Yes @tildeee that sounds good

File: db/seeds.rb (hunk 3 above)
Comment (tildeee) [reply db/seeds.rb]: Just looked at this comment; the seed file actually already keeps track of how many get saved kind of? (It just prints out a count of all `Driver`s)

I've actually never written a seed file before! Would the best way to write this just be:
```
driver_failures = []
...
successful = driver.save
if !successful
  driver_failures << driver
...

puts "Added #{Driver.count} driver records"
puts "#{driver_failures.length} drivers failed to save"
```

?

Comment (kariabancroft) [reply db/seeds.rb]: Yes something like that is good! This is one example from what we did with books seeds for C7 --
 just another way to indicate success 

https://github.com/Ada-C7/stacks-book-app/blob/master/db/seeds.rb

**Pull request merged**